py/client/pydeephaven/session.py

```diff
@@ -118,6 +145,15 @@ def __init__(self, host: str = None, port: int = None, auth_type: str = "Anonymo
         if not port:
             self.port = int(os.environ.get("DH_PORT", 10000))
 
+        self._use_tls = use_tls
+        self._tls_root_certs = tls_root_certs
+        self._client_cert_chain = client_cert_chain
+        self._client_private_key = client_private_key
+        if self._tls_root_certs is not None and not self._use_tls:
+            raise DHError("use_tls is false but tsl_root_certs is not None")
```

Comment:
There are some invalid argument combos not checked. Cristian and I discussed the best way to handle it and we both agreed that no check is probably the best, just let the lower level gRPC throw the exceptions which will be re-thrown as a DHError to the application.